{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#collapse_reads2.py script: had to remove clear_output and sys.argv[1] and sys.exit functions (all calls of \"index\") in order to process lib1_TTGACT.R1.codingrna_aligned file for some reason. added them back. -- they were not loading correctly for some reason and throwing up errors.\n",
    "import csv\n",
    "import gzip\n",
    "import sys\n",
    "import os\n",
    "import cPickle as pickle\n",
    "import subprocess\n",
    "import time\n",
    "import re\n",
    "import pdb\n",
    "\n",
    "index = sys.argv[1]\n",
    "\n",
    " \n",
    "\n",
    "startTime = time.time()\n",
    "\n",
    "countDict = dict()\n",
    "\n",
    " \n",
    "\n",
    "allFiles = os.listdir( os.getcwdu() )\n",
    "\n",
    "alignedFiles = filter( lambda x: re.search( '*codingrna_comb_aligned', x ), allFiles )\n",
    "\n",
    "indices = [ re.search( '_([ACTG]+).R1', x ).group(1) for x in alignedFiles ]\n",
    "\n",
    "if index not in indices:\n",
    "\n",
    "    sys.exit()\n",
    "\n",
    " \n",
    "\n",
    "readCount = 0\n",
    "\n",
    "for currentFile in alignedFiles:\n",
    "\n",
    "    if currentFile.find(index) == -1:\n",
    "\n",
    "       continue\n",
    "\n",
    "    openedFile = open( currentFile )\n",
    "\n",
    "    reader = csv.DictReader( openedFile, delimiter='\\t',\n",
    "\n",
    "                        fieldnames=['readName', 'strand', 'ref', 'pos', 'seq',\n",
    "\n",
    "                                    'qual', 'Mparam', 'mismatchString'] )\n",
    "\n",
    "    numberOfBasesToTrim = 8\n",
    "\n",
    "    for read in reader:\n",
    "\n",
    "        readCount += 1\n",
    "\n",
    "\n",
    "\n",
    "        readLength = len( read['seq'] )\n",
    "\n",
    "        numberOfMismatches = read['mismatchString'].count(':')\n",
    "\n",
    " \n",
    "\n",
    "        if ( numberOfMismatches<3 and readLength>2*numberOfBasesToTrim ): #length and mismatch cutoff.\n",
    "\n",
    "            readStart = int( read['pos'] ) + numberOfBasesToTrim\n",
    "            readEnd = int( read['pos'] ) + readLength - numberOfBasesToTrim\n",
    "            readWeight = 1.0 / (readLength - 2 * numberOfBasesToTrim)\n",
    " \n",
    "\n",
    "                if read['ref']+read['strand'] not in countDict:\n",
    "    \n",
    "                    countDict[read['ref']+read['strand']] = dict()\n",
    "\n",
    "                for readPos in range(readStart, readEnd):\n",
    "                    if readPos not in countDict[read['ref']+read['strand']]:\n",
    "                        countDict[read['ref']+read['strand']][readPos] = 0.0\n",
    "               \t    countDict[read['ref']+read['strand']][readPos] += readWeight\n",
    "        \n",
    "                if readCount % 100000 == 0:\n",
    "    \n",
    "                    currentTime = time.time() - startTime\n",
    "    \n",
    "                    print str( int( currentTime ) ), 'seconds total.', readCount, 'reads processed.'\n",
    "    \n",
    "                if readCount > 1000000:\n",
    "                    break\n",
    "\n",
    "    openedFile.close()\n",
    "\n",
    " \n",
    "\n",
    "pickle.dump( countDict, open( 'index' + '.pickled', 'wb' ) )\n",
    "#output is LIB_INDEX_6bp.pickled file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#codonstats_mammalian.py code\n",
    "seqFile =  'refseq_canonical.fa'\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import CodonUsage\n",
    "import cPickle as pickle\n",
    "import numpy\n",
    "import sys\n",
    "import os\n",
    "import pandas\n",
    "import re\n",
    "import gzip\n",
    "import tables\n",
    "#from IPython.core.display import clear_output\n",
    "\n",
    "cdsCoordinates = pandas.read_table( 'refseqmrna_cdscoordinates.tab' ).set_index( 'refseqID' )\n",
    "\n",
    "offset = 150 #defines what nts to exlcude from start, stop\n",
    "\n",
    "def process_codons(dna_sequence, gene_name): #for each gene                       \n",
    "    # make the codon dictionary local\n",
    "    codon_count = CodonUsage.CodonsDict.copy()\n",
    "    codon_count[\"start\"] = 0\n",
    "    codon_number_dict = CodonUsage.CodonsDict.copy()\n",
    "    for codon_number, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "        codon_number_dict[codon] = codon_number\n",
    "    codon_number_dict[\"start\"] = 64   #this goes through codons and assigns them a number, w/ start as #64\n",
    "    \n",
    "    codon_list = numpy.empty(len(dna_sequence))\n",
    "    codon_list.fill(numpy.nan)  #returns boolean array with \"true\" if value is NaN, otherwise \"false\"\n",
    "    \n",
    "    for i in range(0,len(dna_sequence),3): #i goes from 0 to length of gene in jumps of 3\n",
    "        codon = str( dna_sequence[i:i+3] ).upper()  #defines each codon as a string\n",
    "        if codon in codon_count:            \n",
    "            if i == 0:\n",
    "                codon_list[i] = 64 #first codon is start\n",
    "                codon_count[\"start\"] += 1 #counts 1 codon for each group of 3\n",
    "            else:\n",
    "                codon_count[codon] += 1\n",
    "                codon_list[i] = codon_number_dict[codon]\n",
    "        else:\n",
    "            print gene_name\n",
    "            raise TypeError(\"illegal codon in this gene %s\" % dna_sequence)            \n",
    "    \n",
    "    return (codon_count, codon_list)\n",
    "\n",
    "\n",
    "def process_ribosome_profiling_data( ribosome_profiling_counts, codonList, offset ):    \n",
    "    \n",
    "    total_counts = ribosome_profiling_counts[offset:-offset].sum() # calculate the total counts in each gene's coding sequence.\n",
    "    total_firstthird_counts = ribosome_profiling_counts[offset:-offset].sum()\n",
    "    total_lastthird_counts = ribosome_profiling_counts[offset:-offset].sum()\n",
    "    \n",
    "    codon_number_dict = CodonUsage.CodonsDict.copy() # create local copy of codon dictionary.\n",
    "    \n",
    "    for codon_number, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "        codon_number_dict[codon] = codon_number\n",
    "    codon_number_dict[\"start\"] = 64 # assign numbers to different codons alphabetically.\n",
    "    \n",
    "    codon_ribosome_occupancy = numpy.zeros( (1, len( codon_number_dict ), 2*offset+1 ) ) #initialize count matrix\n",
    "    \n",
    "    codon_ribosome_occupancy_flag = numpy.zeros( (1, len( codon_number_dict ) ), dtype = bool ) #initialize flag matrix     \n",
    "        \n",
    "    for codon_number in codon_number_dict.itervalues(): # iterate over all codons\n",
    "        \n",
    "        locations = numpy.nonzero( codonList == codon_number )[0]\n",
    "        average_counts = ribosome_profiling_counts[offset:-offset].mean()\n",
    "        average_firstthird_counts = \n",
    "        average_lastthird_counts =\n",
    "        \n",
    "        \n",
    "        if len(locations) and average_counts >= 1.0/3.0: # criterion is presence of a codon AND minimum of 1 count per codon\n",
    "            \n",
    "            for codon_location in locations: # iterate over all occurrences of codon and count ribsoome counts.                                \n",
    "                try:\n",
    "                    codon_ribosome_occupancy[0,codon_number,:] = (codon_ribosome_occupancy[0,codon_number,:]\n",
    "                                                                        + ribosome_profiling_counts[codon_location:codon_location+2*offset+1])\n",
    "                except ValueError:                    \n",
    "                    print codon_location\n",
    "                    raise ValueError\n",
    "            codon_ribosome_occupancy[0,codon_number,:] = codon_ribosome_occupancy[0,codon_number,:] / len(locations) / average_counts\n",
    "            codon_ribosome_occupancy_flag[0,codon_number] = 1\n",
    "    \n",
    "    return (codon_ribosome_occupancy, codon_ribosome_occupancy_flag)\n",
    "\n",
    "def calculate_intragenic_profile( ribosome_profiling_counts, codonList, offset, profile, profile_counts ):    \n",
    "    \n",
    "    total_counts = ribosome_profiling_counts[offset:-offset].sum() # calculate the total counts in each gene's coding sequence.\n",
    "    \n",
    "    codon_number_dict = CodonUsage.CodonsDict.copy() # create local copy of codon dictionary.\n",
    "    \n",
    "    for codon_number, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "        codon_number_dict[codon] = codon_number\n",
    "    codon_number_dict[\"start\"] = 64 # assign numbers to different codons alphabetically.    \n",
    "        \n",
    "    for codon_number in codon_number_dict.itervalues(): # iterate over all codons\n",
    "        \n",
    "        locations = numpy.nonzero( codonList == codon_number )[0]\n",
    "        average_counts = ribosome_profiling_counts[offset:-offset].mean()\n",
    "        \n",
    "        if len(locations) and average_counts >= 1.0/3.0: # criterion is presence of a codon AND minimum of 1 count per codon\n",
    "            \n",
    "            for codon_location in filter( lambda x: x < len( profile ) - 2, locations ): # iterate over all occurrences of codon and count ribsoome counts.\n",
    "                profile[ codon_location:codon_location+3 ] = ( profile[ codon_location:codon_location+3]\n",
    "                                                                        + ribosome_profiling_counts[codon_location + offset:codon_location+3+offset] / average_counts )                \n",
    "                profile_counts[ codon_location:codon_location+3 ] = profile_counts[ codon_location:codon_location+3 ] + 1\n",
    "    \n",
    "    return (profile, profile_counts)\n",
    "\n",
    "allFiles = os.listdir('/n/osheafs1/LAB/alicia/data/htseq/20160303_293TAAVS1crispr_hrGFP-ragbwt-ragbq99l_richleuargtorin1_pluslinkerlibs/Lane2/comb_coding_algined/')\n",
    "pickledFiles = filter( lambda x: x.endswith( '*.pickled' ), allFiles )\n",
    "\n",
    "\n",
    "for File in pickledFiles:    \n",
    " \n",
    "    sampleName = '_' + File.split( '.' )[0].lower()\n",
    "    \n",
    "    print 'sample: ' + sampleName\n",
    "    \n",
    "    sys.stdout.flush()\n",
    "    \n",
    "    rpData = pickle.load( open( File ) )\n",
    "\n",
    "    genes = dict()    \n",
    "    \n",
    "            \n",
    "    profile = numpy.zeros( (1000,) )\n",
    "    profile_counts = numpy.zeros( (1000,) )\n",
    "    \n",
    "    rowCounts = 0    \n",
    "    \n",
    "    \n",
    "    with tables.openFile('storage' + sampleName + '.h5',\"w\") as storage:\n",
    "        storageGroup = storage.root\n",
    "        occupancyArray = storage.createEArray(storageGroup,\n",
    "                                              name = \"codonOccupancy\",\n",
    "                                              atom = tables.Float64Atom(dflt=0.0),\n",
    "                                              shape = (0,65, 2*offset+1),\n",
    "                                              expectedrows = 20000\n",
    "                                              )\n",
    "    \n",
    "        occupancyFlagArray = storage.createEArray(storageGroup,\n",
    "                                              name = \"codonOccupancyFlag\",\n",
    "                                              atom = tables.BoolAtom(dflt=False),\n",
    "                                              shape = (0, 65),\n",
    "                                              expectedrows = 20000\n",
    "                                              )\n",
    "        rpCountsStorage = open( 'rpCounts' + sampleName + '.txt', 'w' )\n",
    "        rpCountsStorage.write( 'refseqID' + '\\t' + 'transcriptLength' + '\\t' + 'cdsStart' + '\\t' + 'cdsEnd' + '\\t' + 'counts' + '\\n' )\n",
    "        \n",
    "        for record in SeqIO.parse( open( seqFile ), 'fasta' ):     \n",
    "            if record.id +'+' not in rpData:            \n",
    "                continue \n",
    "            if record.id in ['NM_004152.2','NM_002537.2','NM_016178.2', 'NM_015068.3', 'NM_001184961.1']: #OAZ1, OAZ2, OAZ3, PEG10, PEG10\n",
    "                continue\n",
    "                \n",
    "            start = int( cdsCoordinates.ix[record.id]['cdsStart'] )\n",
    "            \n",
    "            end = int( cdsCoordinates.ix[record.id]['cdsEnd'] )\n",
    "            \n",
    "            seq = record.seq[start:end]\n",
    "            \n",
    "            length = len(seq)        \n",
    "                        \n",
    "    \n",
    "            rpCounts = numpy.zeros( ( len( record.seq ), ) )  \n",
    "            \n",
    "            \n",
    "            for pos in rpData[ record.id +'+' ]:\n",
    "               \n",
    "                rpCounts[ pos ] = rpData[ record.id +'+' ][ pos ]\n",
    "   \n",
    "            geneRpCounts = rpCounts[start - offset :end + offset]\n",
    "            \n",
    "            if start - offset < 0:\n",
    "                geneRpCounts = numpy.hstack( [numpy.zeros( ( offset-start, ) ), rpCounts[:end + offset] ] )    \n",
    "            if end + offset >= len(rpCounts):\n",
    "                geneRpCounts = numpy.hstack( [rpCounts[start - offset : ], numpy.zeros( ( offset + end - len(rpCounts), ) ) ] )    \n",
    "            if start - offset < 0 and end + offset >= len(rpCounts):\n",
    "                geneRpCounts = numpy.hstack( [numpy.zeros( ( offset-start, ) ), rpCounts, numpy.zeros( ( offset + end - len(rpCounts), ) ) ] )    \n",
    "    \n",
    "            ( codonCounts, codonList ) = process_codons( seq, record.id )\n",
    "            \n",
    "            (codon_ribosome_occupancy, codon_ribosome_occupancy_flag) = process_ribosome_profiling_data( geneRpCounts, codonList, offset )\n",
    "            \n",
    "            occupancyArray.append( codon_ribosome_occupancy )\n",
    "            occupancyFlagArray.append( codon_ribosome_occupancy_flag )\n",
    "            \n",
    "            (profile, profile_counts) = calculate_intragenic_profile( geneRpCounts, codonList, offset, profile, profile_counts )\n",
    "                \n",
    "            \n",
    "            \n",
    "            genes[ record.id ] = dict( [ ('start',start), ('end',end), ('seq',seq), ('length', length), \\\n",
    "                                        ('totalCountsTranscript', rpCounts.sum()), ('totalCountsCds', geneRpCounts.sum()), ('readDensityPerNt', geneRpCounts.sum() / length ) ] )       \n",
    "            \n",
    "            rpCountsStorage.write( record.id + '\\t' + str(length) + '\\t' + str(start) + '\\t' + str(end) + '\\t' )\n",
    "            numpy.savetxt( rpCountsStorage, rpCounts.reshape((1,rpCounts.shape[0])), fmt='%0.2f', delimiter='\\t' )\n",
    "                                        \n",
    "            \n",
    "            rowCounts += 1\n",
    "            \n",
    "            if numpy.mod( rowCounts, 200 ) == 0:\n",
    "                print sampleName, rowCounts\n",
    "                sys.stdout.flush()\n",
    "\n",
    "    pandas.save( genes, 'geneStats' + sampleName + '.pickled')   \n",
    "    \n",
    "    sampleStats = dict()      \n",
    "    \n",
    "    sampleStats[ 'profile' ] = profile\n",
    "    sampleStats[ 'profile_counts' ] = profile_counts\n",
    "    sampleStats[ 'profile_density' ] = numpy.divide( profile, profile_counts )\n",
    "    sampleStats[ 'codonRibosomeOccupancyMedian' ] = dict()\n",
    "    sampleStats[ 'codonRibosomeOccupancy25' ] = dict()\n",
    "    sampleStats[ 'codonRibosomeOccupancy75' ] = dict()\n",
    "    sampleStats[ 'codonRibosomeOccupancyMean'] = dict()\n",
    "    sampleStats[ 'codonRibosomeOccupancyMedian_FirstThird'] = dict()\n",
    "    sampleStats[ 'codonRibosomeOccupancyMedian_LastThird'] = dict()\n",
    "    \n",
    "    codon_number_dict = CodonUsage.CodonsDict.copy() # create local copy of codon dictionary.\n",
    "    \n",
    "    for codon_number, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "        codon_number_dict[codon] = codon_number\n",
    "    codon_number_dict[\"start\"] = 64 # assign numbers to different codons alphabetically.\n",
    "    \n",
    "    storage = tables.openFile( 'storage' + sampleName + '.h5', 'r' )\n",
    "    occupancyArray = storage.root.codonOccupancy\n",
    "    occupancyFlagArray = storage.root.codonOccupancyFlag\n",
    "    \n",
    "    for codon, codonindex in codon_number_dict.items():\n",
    "        print codon\n",
    "        nontrivialGenes = occupancyFlagArray[:,codonindex].nonzero()[0]         \n",
    "        ( sampleStats[ 'codonRibosomeOccupancy25' ][ codon ], \\\n",
    "        sampleStats[ 'codonRibosomeOccupancyMedian' ][ codon ], \\\n",
    "        sampleStats[ 'codonRibosomeOccupancy75' ][ codon ] )= numpy.percentile( occupancyArray[ nontrivialGenes,codonindex, ], q=(25, 50, 75), axis=0)\n",
    "        sampleStats[ 'codonRibosomeOccupancyMean' ][ codon ] = numpy.mean( occupancyArray[ nontrivialGenes,codonindex, ], axis=0)\n",
    "        sampleStats[ 'codonRibosomeOccupancyMedian_FirstThird'][ codon ] = \n",
    "        sampleStats[ 'codonRibosomeOccupancyMedian_LastThird'][ codon ] = \n",
    "\n",
    "    storage.close()            \n",
    "    os.remove( 'storage' + sampleName + '.h5' )    \n",
    "    pickle.dump( sampleStats, open( 'sampleStats' + sampleName + '.pickled', 'wb' ) )\n",
    "    \n",
    "    #output is geneStats_index and sampleStats_index.pickled files. scp these to local linux computer and launch ipytho notebook for analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#based on anaylsis from 20140830 ipynb nb.\n",
    "#Start here once codon stats has been run on the cluster \n",
    "#script to process all genes into a single dataframe, run from /Users/aliciadarnell \n",
    "#This needs to be run to generate \"data\" dictionary before I use the following code to analyze results.\n",
    "import cPickle as pickle\n",
    "import re\n",
    "import pandas\n",
    "import sys\n",
    "import numpy\n",
    "from IPython.core.display import clear_output\n",
    "\n",
    "processedFiles = !ls ../geneStats_*.pickled\n",
    "\n",
    "df = list()\n",
    "\n",
    "for File in processedFiles:\n",
    "    #print File\n",
    "    temp = pandas.DataFrame.from_dict(  pickle.load( open( File, 'rb' ) ), orient='index' )                            \n",
    "    sample = re.search( 'geneStats(.*)\\.pickled', File ).groups( 1 )[0]                \n",
    "    print sample\n",
    "    \n",
    "    if len( df ) > 0:\n",
    "        df.append( temp[['totalCountsCds','readDensityPerNt']].rename( columns = {'totalCountsCds': 'counts' + sample, 'readDensityPerNt':'rpnt' + sample} ) )\n",
    "    else:\n",
    "        df.append( temp.drop( ['seq','totalCountsTranscript'], axis = 1).rename( columns = {'totalCountsCds': 'counts' + sample, 'readDensityPerNt':'rpnt' + sample} ) )\n",
    "        \n",
    "    print File\n",
    "    sys.stdout.flush()\n",
    "    \n",
    "annotations = pandas.load( '../seq_annotation_files/refseq_annotations.pickled' )\n",
    "df.append( annotations )\n",
    "        \n",
    "data = pandas.concat( df, join='inner', axis=1 )\n",
    "\n",
    "for key in data.keys():\n",
    "    if key.startswith( 'counts' ):\n",
    "        data[ key ] = data[ key ].apply( lambda x: numpy.round( x, 0) )                \n",
    "    if key.startswith( 'rpnt' ):\n",
    "        data[ key ] = data[ key ].apply( lambda x: numpy.round( x, 2) )                \n",
    "\n",
    "data['description'] = map( lambda x: x.replace( 'Homo sapiens ', '' ), data['description'] )\n",
    "    \n",
    "clear_output()\n",
    "data['gene'] = map( lambda x: re.search( '\\(([\\w\\-]+)\\),', x ).groups()[0], data['description'] )\n",
    "data = data.reset_index().rename( columns = {'index':'refseqid'} ).set_index( 'gene' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import cPickle as pickle\n",
    "%matplotlib inline\n",
    "top_samples = ['hrgfp_arg','ragbwt_arg','ragbq99l_arg','hrgfp_torin1arg']#,'hrgfp_torin1leu','hrgfp_torin1arg','hrgfp_linker_leu','ragbq99l_linker_leu']\n",
    "bottom_samples = ['hrgfp_rich','ragbwt_rich','ragbq99l_rich','hrgfp_torin1rich']#,'hrgfp_torin1rich','hrgfp_torin1rich','hrgfp_linker_rich','hrgfp_linker_rich']\n",
    "\n",
    "offset = 150\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "from matplotlib.ticker import MultipleLocator, NullLocator, MaxNLocator\n",
    "from Bio.SeqUtils import CodonUsage\n",
    "from matplotlib.transforms import Bbox\n",
    "import numpy\n",
    "\n",
    "\n",
    "codon_number_dict = CodonUsage.CodonsDict.copy() # create local copy of codon dictionary.\n",
    "    \n",
    "for codon_number, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "    codon_number_dict[codon] = codon_number\n",
    "codon_number_dict[\"start\"] = 64 # assign numbers to different codons alphabetically.\n",
    "\n",
    "plt.rcParams['pdf.fonttype'] = 42\n",
    "plt.rcParams['font.family'] = 'sans-serif'\n",
    "plt.rcParams['font.sans-serif'] = 'Liberation Sans'\n",
    "plt.rcParams['font.size'] = 14\n",
    "\n",
    "fig = plt.figure()\n",
    "common_ax = plt.axes(frameon=False)\n",
    "common_ax.set_xlabel('Distance from Codon (nt)', fontsize = \"large\")\n",
    "common_ax.set_ylabel(r'Starved Read Density $\\minus$ Rich Read Density', fontsize = \"large\")\n",
    "common_ax.set_xticks([])\n",
    "common_ax.set_yticks([])\n",
    "common_ax.yaxis.labelpad = 40\n",
    "common_ax.xaxis.labelpad = 30First\n",
    "\n",
    "fig.subplots_adjust( wspace=0.5, hspace=0.5 )\n",
    "\n",
    "edge_colors = [\"chocolate\", \"red\", \"black\", \"green\", \"purple\", \"darkslateblue\"]\n",
    "summedDensity = dict()\n",
    "stallingIndex = dict()\n",
    "for sampleIndex, sample in enumerate( top_samples ):\n",
    "    sampleStats_top = pickle.load( open( '../sampleStats_' + top_samples[sampleIndex] + '.pickled', 'rb' ) )\n",
    "    sampleStats_bottom = pickle.load( open( '../sampleStats_' + bottom_samples[sampleIndex] + '.pickled', 'rb' ) )\n",
    "    \n",
    "    summedDensity[top_samples[sampleIndex]] = dict()    \n",
    "    if re.search( 'hrgfp_leu', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['TTA', 'TTG', 'CTT', 'CTC', 'CTG', 'CTA']\n",
    "        legendTitle = 'Leu Codons'\n",
    "        axTitle = '-Leucine, hrGFP'\n",
    "    elif re.search( 'hrgfp_arg', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['AGA', 'AGG', 'CGA', 'CGC', 'CGG', 'CGT']\n",
    "        legendTitle = 'Arg Codons'\n",
    "        axTitle = '-Arginine, hrGFP'\n",
    "    elif re.search( 'ragbwt_leu', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['TTA', 'TTG', 'CTT', 'CTC', 'CTG', 'CTA']\n",
    "        legendTitle = 'Leu Codons'\n",
    "        axTitle = '-Leucine, RagB WT'\n",
    "    elif re.search( 'ragbwt_arg', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['AGA', 'AGG', 'CGA', 'CGC', 'CGG', 'CGT']\n",
    "        legendTitle = 'Arg Codons'\n",
    "        axTitle = '-Arginine, RagB WT'\n",
    "    elif re.search( 'ragbq99l_leu', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['TTA', 'TTG', 'CTT', 'CTC', 'CTG', 'CTA']\n",
    "        legendTitle = 'Leu Codons'\n",
    "        axTitle = '-Leucine, RagB Q99L'\n",
    "    elif re.search( 'ragbq99l_arg', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['AGA', 'AGG', 'CGA', 'CGC', 'CGG', 'CGT']\n",
    "        legendTitle = 'Arg Codons'\n",
    "        axTitle = '-Arginine, RagB Q99L'\n",
    "    elif re.search( 'hrgfp_torin1leu', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['TTA', 'TTG', 'CTT', 'CTC', 'CTG', 'CTA']\n",
    "        legendTitle = 'Leu Codons'\n",
    "        axTitle = '-Leucine + Torin1, hrGFP'\n",
    "    elif re.search( 'hrgfp_torin1arg', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['AGA', 'AGG', 'CGA', 'CGC', 'CGG', 'CGT']\n",
    "        legendTitle = 'Arg Codons'First\n",
    "        axTitle = '-Arginine + Torin1, hrGFP'\n",
    "    elif re.search( 'hrgfp_linker_leu', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['TTA', 'TTG', 'CTT', 'CTC', 'CTG', 'CTA']\n",
    "        legendTitle = 'Leu Codons'\n",
    "        axTitle = '-Leucine, hrGFP Linker Prep'\n",
    "    elif re.search( 'ragbq99l_linker_leu', top_samples[sampleIndex] ):\n",
    "        subsetCodons = ['TTA', 'TTG', 'CTT', 'CTC', 'CTG', 'CTA']\n",
    "        legendTitle = 'Leu Codons'\n",
    "        axTitle = '-Leucine, RagBQ99L Linker Prep'\n",
    "        \n",
    "    \n",
    "    ax = fig.add_subplot( int(numpy.ceil(len(top_samples)/2.0)),2,sampleIndex+1)     \n",
    "    temp = list()\n",
    "    \n",
    "    for codonIndex, codon in enumerate( sorted( filter( lambda codon: codon in subsetCodons, codon_number_dict ) ) ):\n",
    "        x_vector = numpy.arange(-offset,+offset+1)\n",
    "        y_vector = sampleStats_top['codonRibosomeOccupancyMedian'][codon] - sampleStats_bottom['codonRibosomeOccupancyMedian'][codon]\n",
    "        \n",
    "        ax.plot(x_vector, y_vector, color = edge_colors[codonIndex], label=codon, linewidth=2, marker=None);        \n",
    "        #print sample, codon, y_vector[offset-30:offset+50].sum() - min(temp) + 1    \n",
    "    \n",
    "    for loc, spine in ax.spines.items():\n",
    "        if loc in ['left','bottom']:\n",
    "            spine.set_position(('outward',10)) # outward by 10 points\n",
    "        elif loc in ['right','top']:\n",
    "            spine.set_color('none') # don't draw spine\n",
    "        else:\n",
    "            raise ValueError('unknown spine location: %s'%loc)\n",
    "\n",
    "    # turn off ticks where there is no spine\n",
    "    ax.xaxis.set_ticks_position('bottom')\n",
    "    ax.yaxis.set_ticks_position('left')\n",
    "    ax.xaxis.set_major_locator(MaxNLocator(7))\n",
    "    ax.xaxis.set_ticks_position('bottom')\n",
    "    ax.set_ylim(bottom=-0.35, top=1)\n",
    "    ax.yaxis.set_major_locator(MaxNLocator(5, prune=None))\n",
    "    ax.yaxis.set_ticks_position('left')\n",
    "    ax.set_xlim(-offset,+offset)\n",
    "    ax.xaxis.grid()\n",
    "    \n",
    "    ax.set_title( axTitle )\n",
    "    temp = ax.legend(frameon=False, title=legendTitle)\n",
    "    temp.get_title().set_fontweight( 'bold' )\n",
    "    \n",
    "    bboxvalues = ax.get_position().get_points()\n",
    "    left = bboxvalues[0,0] + 0.03\n",
    "    bottom = bboxvalues[0,1] + 0.17  #adjust this value if the height of the box is off-- higher makes it go up relative to its plot and lower makes it go down.\n",
    "    width = 0.35* ( bboxvalues[1,0]-bboxvalues[0,0] )\n",
    "    height = 0.35* ( bboxvalues[1,1]-bboxvalues[0,1] )\n",
    "    \n",
    "    ax2 = fig.add_axes([left,bottom,width,height])\n",
    "    \n",
    "    for codonIndex, codon in enumerate( sorted( filter( lambda codon: codon not in subsetCodons, codon_number_dict ) ) ):\n",
    "        x_vector = numpy.arange(-offset,+offset+1)\n",
    "        \n",
    "        if codon in ['TGA','TAA','TAG','start']:\n",
    "            continue\n",
    "        yvector = sampleStats_top['codonRibosomeOccupancyMedian'][codon] - sampleStats_bottom['codonRibosomeOccupancyMedian'][codon]\n",
    "        \n",
    "        \n",
    "        ax2.plot(x_vector, yvector,    \n",
    "            color = edge_colors[numpy.mod(codonIndex,6)], label=codon, linewidth=0.5, marker=None);\n",
    "        \n",
    "    ax2.set_xlim(ax.get_xlim())\n",
    "    ax2.set_ylim(ax.get_ylim())\n",
    "    [i.set_fontsize( 'small' ) for i in ax2.get_yticklabels()]\n",
    "    ax2.xaxis.set_major_locator(ax.xaxis.get_major_locator())    \n",
    "    ax2.yaxis.set_major_locator(ax.yaxis.get_major_locator())    \n",
    "    ax2.xaxis.set_ticklabels([])\n",
    "    #ax2.yaxis.set_ticklabels([])\n",
    "    ax2.set_title( 'All other codons', fontsize='small')  \n",
    "    \n",
    "fig.set_size_inches([12,4 * numpy.ceil(len(top_samples)/2.0)])\n",
    "fig.savefig( '../plots/hrgfp-ragbs_arg.svg' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import cPickle as pickle\n",
    "%pylab inline\n",
    "offset = 50   \n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "from matplotlib.ticker import MultipleLocator, NullLocator, MaxNLocator\n",
    "from Bio.SeqUtils import CodonUsage\n",
    "\n",
    "codon_number_dict = CodonUsage.CodonsDict.copy() # create local copy of codon dictionary.\n",
    "    \n",
    "for codon_number, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "    codon_number_dict[codon] = codon_number\n",
    "codon_number_dict[\"start\"] = 64 # assign numbers to different codons alphabetically.\n",
    "\n",
    "plt.rcParams['pdf.fonttype'] = 42\n",
    "plt.rcParams['font.family'] = 'sans-serif'\n",
    "plt.rcParams['font.sans-serif'] = 'Arial'\n",
    "plt.rcParams['font.size'] = 12\n",
    "\n",
    "fig = plt.figure()\n",
    "common_ax = plt.axes(frameon=False)\n",
    "common_ax.set_xlabel('Distance from Codon (nt)', fontsize = \"larger\")\n",
    "common_ax.set_ylabel(r'Read Density', fontsize = \"larger\")\n",
    "common_ax.set_xticks([])\n",
    "common_ax.set_yticks([])\n",
    "common_ax.yaxis.labelpad = 40\n",
    "common_ax.xaxis.labelpad = 30\n",
    "\n",
    "sampleStats = pickle.load( open('../sampleStats_ragbwt_leu.pickled'))\n",
    "\n",
    "for codonIndex, codon in enumerate( sorted( codon_number_dict ) ):\n",
    "    x_vector = numpy.arange(-offset,+offset+1)\n",
    "    ax = fig.add_subplot(17,4,codonIndex+1)\n",
    "    plt.fill_between(x_vector, sampleStats['codonRibosomeOccupancyMedian'][codon][offset+50:+offset+150+1],        \n",
    "        edgecolor = 'black', facecolor = 'slategrey' );\n",
    "    plt.plot(x_vector, sampleStats['codonRibosomeOccupancy25'][codon][offset+50:+offset+150+1], \":k\");\n",
    "    plt.plot(x_vector, sampleStats['codonRibosomeOccupancy75'][codon][offset+50:+offset+150+1], \":k\");    \n",
    "    ax.set_title(codon, x=0.02, y=0.7, ha = 'left', fontweight=\"bold\");\n",
    "    if codonIndex < 62:\n",
    "        ax.xaxis.set_ticklabels([])\n",
    "    ax.xaxis.set_major_locator(MaxNLocator(8))\n",
    "    ax.xaxis.set_ticks_position('bottom')\n",
    "    ax.set_ylim(bottom=0)\n",
    "    ax.yaxis.set_major_locator(MaxNLocator(3, prune=None))\n",
    "    ax.yaxis.set_ticks_position('left')\n",
    "    ax.set_xlim(-offset,+offset)#plt.ylim((0,max([plt.ylim()[1],2])));\n",
    "    ax.xaxis.grid()\n",
    "fig.set_size_inches([18,20])\n",
    "\n",
    "#this is a series of 64 plots of the ribosome occupancy (normalized occupancy counts) across all possible codons - look from a certain offet ,\n",
    "#also the dotted lines are quartile boundaries (75% of the data falls under top line, for example). Can change the sample, the offset, *make sure the locations of the \n",
    "#pickled file in sampleStats is correct. If the offset is changed, make the corresponding change later in the codon [never change x_vector, just change offset and adjust scale when plotting with plt. ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refseqid</th>\n",
       "      <th>rpnt_hrgfp_arg</th>\n",
       "      <th>end</th>\n",
       "      <th>start</th>\n",
       "      <th>length</th>\n",
       "      <th>counts_hrgfp_arg</th>\n",
       "      <th>counts_hrgfp_leu</th>\n",
       "      <th>rpnt_hrgfp_leu</th>\n",
       "      <th>counts_hrgfp_linker_leu</th>\n",
       "      <th>rpnt_hrgfp_linker_leu</th>\n",
       "      <th>counts_hrgfp_linker_rich</th>\n",
       "      <th>rpnt_hrgfp_linker_rich</th>\n",
       "      <th>counts_hrgfp_rich</th>\n",
       "      <th>rpnt_hrgfp_rich</th>\n",
       "      <th>counts_hrgfp_torin1arg</th>\n",
       "      <th>rpnt_hrgfp_torin1arg</th>\n",
       "      <th>counts_hrgfp_torin1leu</th>\n",
       "      <th>rpnt_hrgfp_torin1leu</th>\n",
       "      <th>counts_hrgfp_torin1rich</th>\n",
       "      <th>rpnt_hrgfp_torin1rich</th>\n",
       "      <th>counts_ragbq99l_arg</th>\n",
       "      <th>rpnt_ragbq99l_arg</th>\n",
       "      <th>counts_ragbq99l_leu</th>\n",
       "      <th>rpnt_ragbq99l_leu</th>\n",
       "      <th>counts_ragbq99l_linker_leu</th>\n",
       "      <th>rpnt_ragbq99l_linker_leu</th>\n",
       "      <th>counts_ragbq99l_rich</th>\n",
       "      <th>rpnt_ragbq99l_rich</th>\n",
       "      <th>counts_ragbwt_arg</th>\n",
       "      <th>rpnt_ragbwt_arg</th>\n",
       "      <th>counts_ragbwt_leu</th>\n",
       "      <th>rpnt_ragbwt_leu</th>\n",
       "      <th>counts_ragbwt_rich</th>\n",
       "      <th>rpnt_ragbwt_rich</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MTRNR2L4</th>\n",
       "      <td>NM_001190476.1</td>\n",
       "      <td>33.90</td>\n",
       "      <td>468</td>\n",
       "      <td>381</td>\n",
       "      <td>87</td>\n",
       "      <td>2949</td>\n",
       "      <td>1328</td>\n",
       "      <td>15.26</td>\n",
       "      <td>32</td>\n",
       "      <td>0.37</td>\n",
       "      <td>19</td>\n",
       "      <td>0.22</td>\n",
       "      <td>1088</td>\n",
       "      <td>12.51</td>\n",
       "      <td>3213</td>\n",
       "      <td>36.93</td>\n",
       "      <td>4333</td>\n",
       "      <td>49.80</td>\n",
       "      <td>3028</td>\n",
       "      <td>34.80</td>\n",
       "      <td>10262</td>\n",
       "      <td>117.95</td>\n",
       "      <td>5680</td>\n",
       "      <td>65.29</td>\n",
       "      <td>33</td>\n",
       "      <td>0.38</td>\n",
       "      <td>5216</td>\n",
       "      <td>59.95</td>\n",
       "      <td>4701</td>\n",
       "      <td>54.03</td>\n",
       "      <td>3484</td>\n",
       "      <td>40.05</td>\n",
       "      <td>1831</td>\n",
       "      <td>21.05</td>\n",
       "      <td>MT-RNR2-like 4 (MTRNR2L4), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS4X</th>\n",
       "      <td>NM_001007.4</td>\n",
       "      <td>4.80</td>\n",
       "      <td>888</td>\n",
       "      <td>96</td>\n",
       "      <td>792</td>\n",
       "      <td>3798</td>\n",
       "      <td>3172</td>\n",
       "      <td>4.01</td>\n",
       "      <td>1820</td>\n",
       "      <td>2.30</td>\n",
       "      <td>2521</td>\n",
       "      <td>3.18</td>\n",
       "      <td>4766</td>\n",
       "      <td>6.02</td>\n",
       "      <td>1900</td>\n",
       "      <td>2.40</td>\n",
       "      <td>1810</td>\n",
       "      <td>2.29</td>\n",
       "      <td>1798</td>\n",
       "      <td>2.27</td>\n",
       "      <td>5912</td>\n",
       "      <td>7.46</td>\n",
       "      <td>4676</td>\n",
       "      <td>5.90</td>\n",
       "      <td>3050</td>\n",
       "      <td>3.85</td>\n",
       "      <td>5064</td>\n",
       "      <td>6.39</td>\n",
       "      <td>5129</td>\n",
       "      <td>6.48</td>\n",
       "      <td>4161</td>\n",
       "      <td>5.25</td>\n",
       "      <td>4785</td>\n",
       "      <td>6.04</td>\n",
       "      <td>ribosomal protein S4, X-linked (RPS4X), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS13</th>\n",
       "      <td>NM_001017.2</td>\n",
       "      <td>3.38</td>\n",
       "      <td>488</td>\n",
       "      <td>32</td>\n",
       "      <td>456</td>\n",
       "      <td>1540</td>\n",
       "      <td>1559</td>\n",
       "      <td>3.42</td>\n",
       "      <td>788</td>\n",
       "      <td>1.73</td>\n",
       "      <td>926</td>\n",
       "      <td>2.03</td>\n",
       "      <td>1999</td>\n",
       "      <td>4.38</td>\n",
       "      <td>794</td>\n",
       "      <td>1.74</td>\n",
       "      <td>840</td>\n",
       "      <td>1.84</td>\n",
       "      <td>705</td>\n",
       "      <td>1.55</td>\n",
       "      <td>2406</td>\n",
       "      <td>5.28</td>\n",
       "      <td>2376</td>\n",
       "      <td>5.21</td>\n",
       "      <td>1341</td>\n",
       "      <td>2.94</td>\n",
       "      <td>1937</td>\n",
       "      <td>4.25</td>\n",
       "      <td>1892</td>\n",
       "      <td>4.15</td>\n",
       "      <td>2063</td>\n",
       "      <td>4.52</td>\n",
       "      <td>2164</td>\n",
       "      <td>4.75</td>\n",
       "      <td>ribosomal protein S13 (RPS13), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EEF1G</th>\n",
       "      <td>NM_001404.4</td>\n",
       "      <td>1.15</td>\n",
       "      <td>1460</td>\n",
       "      <td>146</td>\n",
       "      <td>1314</td>\n",
       "      <td>1511</td>\n",
       "      <td>997</td>\n",
       "      <td>0.76</td>\n",
       "      <td>850</td>\n",
       "      <td>0.65</td>\n",
       "      <td>1556</td>\n",
       "      <td>1.18</td>\n",
       "      <td>1831</td>\n",
       "      <td>1.39</td>\n",
       "      <td>473</td>\n",
       "      <td>0.36</td>\n",
       "      <td>475</td>\n",
       "      <td>0.36</td>\n",
       "      <td>440</td>\n",
       "      <td>0.33</td>\n",
       "      <td>2290</td>\n",
       "      <td>1.74</td>\n",
       "      <td>1710</td>\n",
       "      <td>1.30</td>\n",
       "      <td>1389</td>\n",
       "      <td>1.06</td>\n",
       "      <td>1707</td>\n",
       "      <td>1.30</td>\n",
       "      <td>1759</td>\n",
       "      <td>1.34</td>\n",
       "      <td>1270</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1652</td>\n",
       "      <td>1.26</td>\n",
       "      <td>eukaryotic translation elongation factor 1 gam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS8</th>\n",
       "      <td>NM_001012.1</td>\n",
       "      <td>2.26</td>\n",
       "      <td>650</td>\n",
       "      <td>23</td>\n",
       "      <td>627</td>\n",
       "      <td>1414</td>\n",
       "      <td>1220</td>\n",
       "      <td>1.95</td>\n",
       "      <td>1058</td>\n",
       "      <td>1.69</td>\n",
       "      <td>1856</td>\n",
       "      <td>2.96</td>\n",
       "      <td>2060</td>\n",
       "      <td>3.29</td>\n",
       "      <td>655</td>\n",
       "      <td>1.04</td>\n",
       "      <td>710</td>\n",
       "      <td>1.13</td>\n",
       "      <td>680</td>\n",
       "      <td>1.08</td>\n",
       "      <td>2205</td>\n",
       "      <td>3.52</td>\n",
       "      <td>1560</td>\n",
       "      <td>2.49</td>\n",
       "      <td>1748</td>\n",
       "      <td>2.79</td>\n",
       "      <td>1666</td>\n",
       "      <td>2.66</td>\n",
       "      <td>1868</td>\n",
       "      <td>2.98</td>\n",
       "      <td>1449</td>\n",
       "      <td>2.31</td>\n",
       "      <td>2002</td>\n",
       "      <td>3.19</td>\n",
       "      <td>ribosomal protein S8 (RPS8), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL13A</th>\n",
       "      <td>NM_012423.3</td>\n",
       "      <td>2.07</td>\n",
       "      <td>688</td>\n",
       "      <td>76</td>\n",
       "      <td>612</td>\n",
       "      <td>1269</td>\n",
       "      <td>746</td>\n",
       "      <td>1.22</td>\n",
       "      <td>677</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1665</td>\n",
       "      <td>2.72</td>\n",
       "      <td>1642</td>\n",
       "      <td>2.68</td>\n",
       "      <td>329</td>\n",
       "      <td>0.54</td>\n",
       "      <td>407</td>\n",
       "      <td>0.67</td>\n",
       "      <td>345</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1941</td>\n",
       "      <td>3.17</td>\n",
       "      <td>1350</td>\n",
       "      <td>2.21</td>\n",
       "      <td>1408</td>\n",
       "      <td>2.30</td>\n",
       "      <td>1279</td>\n",
       "      <td>2.09</td>\n",
       "      <td>1676</td>\n",
       "      <td>2.74</td>\n",
       "      <td>992</td>\n",
       "      <td>1.62</td>\n",
       "      <td>1333</td>\n",
       "      <td>2.18</td>\n",
       "      <td>ribosomal protein L13a (RPL13A), transcript va...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL37</th>\n",
       "      <td>NM_000997.4</td>\n",
       "      <td>3.96</td>\n",
       "      <td>394</td>\n",
       "      <td>100</td>\n",
       "      <td>294</td>\n",
       "      <td>1164</td>\n",
       "      <td>825</td>\n",
       "      <td>2.81</td>\n",
       "      <td>518</td>\n",
       "      <td>1.76</td>\n",
       "      <td>1119</td>\n",
       "      <td>3.81</td>\n",
       "      <td>1578</td>\n",
       "      <td>5.37</td>\n",
       "      <td>468</td>\n",
       "      <td>1.59</td>\n",
       "      <td>481</td>\n",
       "      <td>1.64</td>\n",
       "      <td>433</td>\n",
       "      <td>1.47</td>\n",
       "      <td>1775</td>\n",
       "      <td>6.04</td>\n",
       "      <td>1038</td>\n",
       "      <td>3.53</td>\n",
       "      <td>857</td>\n",
       "      <td>2.91</td>\n",
       "      <td>1265</td>\n",
       "      <td>4.30</td>\n",
       "      <td>1482</td>\n",
       "      <td>5.04</td>\n",
       "      <td>889</td>\n",
       "      <td>3.02</td>\n",
       "      <td>1561</td>\n",
       "      <td>5.31</td>\n",
       "      <td>ribosomal protein L37 (RPL37), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS19</th>\n",
       "      <td>NM_001022.3</td>\n",
       "      <td>2.03</td>\n",
       "      <td>810</td>\n",
       "      <td>372</td>\n",
       "      <td>438</td>\n",
       "      <td>889</td>\n",
       "      <td>762</td>\n",
       "      <td>1.74</td>\n",
       "      <td>875</td>\n",
       "      <td>2.00</td>\n",
       "      <td>2008</td>\n",
       "      <td>4.58</td>\n",
       "      <td>1899</td>\n",
       "      <td>4.34</td>\n",
       "      <td>230</td>\n",
       "      <td>0.53</td>\n",
       "      <td>363</td>\n",
       "      <td>0.83</td>\n",
       "      <td>349</td>\n",
       "      <td>0.80</td>\n",
       "      <td>1485</td>\n",
       "      <td>3.39</td>\n",
       "      <td>1331</td>\n",
       "      <td>3.04</td>\n",
       "      <td>1602</td>\n",
       "      <td>3.66</td>\n",
       "      <td>1492</td>\n",
       "      <td>3.41</td>\n",
       "      <td>1243</td>\n",
       "      <td>2.84</td>\n",
       "      <td>1020</td>\n",
       "      <td>2.33</td>\n",
       "      <td>1621</td>\n",
       "      <td>3.70</td>\n",
       "      <td>ribosomal protein S19 (RPS19), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPLP2</th>\n",
       "      <td>NM_001004.3</td>\n",
       "      <td>2.66</td>\n",
       "      <td>453</td>\n",
       "      <td>105</td>\n",
       "      <td>348</td>\n",
       "      <td>927</td>\n",
       "      <td>870</td>\n",
       "      <td>2.50</td>\n",
       "      <td>562</td>\n",
       "      <td>1.61</td>\n",
       "      <td>1042</td>\n",
       "      <td>2.99</td>\n",
       "      <td>1625</td>\n",
       "      <td>4.67</td>\n",
       "      <td>430</td>\n",
       "      <td>1.24</td>\n",
       "      <td>520</td>\n",
       "      <td>1.49</td>\n",
       "      <td>371</td>\n",
       "      <td>1.07</td>\n",
       "      <td>1467</td>\n",
       "      <td>4.22</td>\n",
       "      <td>1392</td>\n",
       "      <td>4.00</td>\n",
       "      <td>1080</td>\n",
       "      <td>3.10</td>\n",
       "      <td>1476</td>\n",
       "      <td>4.24</td>\n",
       "      <td>882</td>\n",
       "      <td>2.53</td>\n",
       "      <td>1363</td>\n",
       "      <td>3.92</td>\n",
       "      <td>2093</td>\n",
       "      <td>6.01</td>\n",
       "      <td>ribosomal protein, large, P2 (RPLP2), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS28</th>\n",
       "      <td>NM_001031.4</td>\n",
       "      <td>3.34</td>\n",
       "      <td>241</td>\n",
       "      <td>31</td>\n",
       "      <td>210</td>\n",
       "      <td>701</td>\n",
       "      <td>417</td>\n",
       "      <td>1.99</td>\n",
       "      <td>223</td>\n",
       "      <td>1.06</td>\n",
       "      <td>336</td>\n",
       "      <td>1.60</td>\n",
       "      <td>570</td>\n",
       "      <td>2.71</td>\n",
       "      <td>183</td>\n",
       "      <td>0.87</td>\n",
       "      <td>218</td>\n",
       "      <td>1.04</td>\n",
       "      <td>149</td>\n",
       "      <td>0.71</td>\n",
       "      <td>1326</td>\n",
       "      <td>6.31</td>\n",
       "      <td>861</td>\n",
       "      <td>4.10</td>\n",
       "      <td>420</td>\n",
       "      <td>2.00</td>\n",
       "      <td>556</td>\n",
       "      <td>2.65</td>\n",
       "      <td>1037</td>\n",
       "      <td>4.94</td>\n",
       "      <td>650</td>\n",
       "      <td>3.10</td>\n",
       "      <td>678</td>\n",
       "      <td>3.23</td>\n",
       "      <td>ribosomal protein S28 (RPS28), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL37A</th>\n",
       "      <td>NM_000998.4</td>\n",
       "      <td>2.68</td>\n",
       "      <td>357</td>\n",
       "      <td>78</td>\n",
       "      <td>279</td>\n",
       "      <td>747</td>\n",
       "      <td>705</td>\n",
       "      <td>2.53</td>\n",
       "      <td>629</td>\n",
       "      <td>2.25</td>\n",
       "      <td>1201</td>\n",
       "      <td>4.30</td>\n",
       "      <td>1622</td>\n",
       "      <td>5.81</td>\n",
       "      <td>256</td>\n",
       "      <td>0.92</td>\n",
       "      <td>373</td>\n",
       "      <td>1.34</td>\n",
       "      <td>310</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1184</td>\n",
       "      <td>4.24</td>\n",
       "      <td>936</td>\n",
       "      <td>3.35</td>\n",
       "      <td>1136</td>\n",
       "      <td>4.07</td>\n",
       "      <td>1235</td>\n",
       "      <td>4.43</td>\n",
       "      <td>1064</td>\n",
       "      <td>3.81</td>\n",
       "      <td>855</td>\n",
       "      <td>3.06</td>\n",
       "      <td>1391</td>\n",
       "      <td>4.99</td>\n",
       "      <td>ribosomal protein L37a (RPL37A), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS17L</th>\n",
       "      <td>NM_001199057.1</td>\n",
       "      <td>1.62</td>\n",
       "      <td>437</td>\n",
       "      <td>29</td>\n",
       "      <td>408</td>\n",
       "      <td>660</td>\n",
       "      <td>751</td>\n",
       "      <td>1.84</td>\n",
       "      <td>429</td>\n",
       "      <td>1.05</td>\n",
       "      <td>769</td>\n",
       "      <td>1.88</td>\n",
       "      <td>1058</td>\n",
       "      <td>2.59</td>\n",
       "      <td>268</td>\n",
       "      <td>0.66</td>\n",
       "      <td>324</td>\n",
       "      <td>0.79</td>\n",
       "      <td>285</td>\n",
       "      <td>0.70</td>\n",
       "      <td>1099</td>\n",
       "      <td>2.69</td>\n",
       "      <td>1210</td>\n",
       "      <td>2.97</td>\n",
       "      <td>710</td>\n",
       "      <td>1.74</td>\n",
       "      <td>961</td>\n",
       "      <td>2.36</td>\n",
       "      <td>955</td>\n",
       "      <td>2.34</td>\n",
       "      <td>942</td>\n",
       "      <td>2.31</td>\n",
       "      <td>1048</td>\n",
       "      <td>2.57</td>\n",
       "      <td>ribosomal protein S17-like (RPS17L), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS17</th>\n",
       "      <td>NM_001021.3</td>\n",
       "      <td>1.64</td>\n",
       "      <td>437</td>\n",
       "      <td>29</td>\n",
       "      <td>408</td>\n",
       "      <td>670</td>\n",
       "      <td>718</td>\n",
       "      <td>1.76</td>\n",
       "      <td>413</td>\n",
       "      <td>1.01</td>\n",
       "      <td>799</td>\n",
       "      <td>1.96</td>\n",
       "      <td>1115</td>\n",
       "      <td>2.73</td>\n",
       "      <td>254</td>\n",
       "      <td>0.62</td>\n",
       "      <td>343</td>\n",
       "      <td>0.84</td>\n",
       "      <td>274</td>\n",
       "      <td>0.67</td>\n",
       "      <td>1016</td>\n",
       "      <td>2.49</td>\n",
       "      <td>1222</td>\n",
       "      <td>3.00</td>\n",
       "      <td>682</td>\n",
       "      <td>1.67</td>\n",
       "      <td>996</td>\n",
       "      <td>2.44</td>\n",
       "      <td>960</td>\n",
       "      <td>2.35</td>\n",
       "      <td>896</td>\n",
       "      <td>2.20</td>\n",
       "      <td>1090</td>\n",
       "      <td>2.67</td>\n",
       "      <td>ribosomal protein S17 (RPS17), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL27A</th>\n",
       "      <td>NM_000990.4</td>\n",
       "      <td>1.35</td>\n",
       "      <td>813</td>\n",
       "      <td>366</td>\n",
       "      <td>447</td>\n",
       "      <td>605</td>\n",
       "      <td>571</td>\n",
       "      <td>1.28</td>\n",
       "      <td>464</td>\n",
       "      <td>1.04</td>\n",
       "      <td>1278</td>\n",
       "      <td>2.86</td>\n",
       "      <td>1403</td>\n",
       "      <td>3.14</td>\n",
       "      <td>160</td>\n",
       "      <td>0.36</td>\n",
       "      <td>189</td>\n",
       "      <td>0.42</td>\n",
       "      <td>192</td>\n",
       "      <td>0.43</td>\n",
       "      <td>994</td>\n",
       "      <td>2.22</td>\n",
       "      <td>1032</td>\n",
       "      <td>2.31</td>\n",
       "      <td>1190</td>\n",
       "      <td>2.66</td>\n",
       "      <td>1008</td>\n",
       "      <td>2.26</td>\n",
       "      <td>900</td>\n",
       "      <td>2.01</td>\n",
       "      <td>811</td>\n",
       "      <td>1.81</td>\n",
       "      <td>1193</td>\n",
       "      <td>2.67</td>\n",
       "      <td>ribosomal protein L27a (RPL27A), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL36A</th>\n",
       "      <td>NM_021029.5</td>\n",
       "      <td>1.27</td>\n",
       "      <td>475</td>\n",
       "      <td>46</td>\n",
       "      <td>429</td>\n",
       "      <td>545</td>\n",
       "      <td>446</td>\n",
       "      <td>1.04</td>\n",
       "      <td>324</td>\n",
       "      <td>0.76</td>\n",
       "      <td>676</td>\n",
       "      <td>1.58</td>\n",
       "      <td>1210</td>\n",
       "      <td>2.82</td>\n",
       "      <td>203</td>\n",
       "      <td>0.47</td>\n",
       "      <td>259</td>\n",
       "      <td>0.60</td>\n",
       "      <td>198</td>\n",
       "      <td>0.46</td>\n",
       "      <td>830</td>\n",
       "      <td>1.93</td>\n",
       "      <td>666</td>\n",
       "      <td>1.55</td>\n",
       "      <td>596</td>\n",
       "      <td>1.39</td>\n",
       "      <td>691</td>\n",
       "      <td>1.61</td>\n",
       "      <td>759</td>\n",
       "      <td>1.77</td>\n",
       "      <td>664</td>\n",
       "      <td>1.55</td>\n",
       "      <td>1007</td>\n",
       "      <td>2.35</td>\n",
       "      <td>ribosomal protein L36a (RPL36A), transcript va...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL36A-HNRNPH2</th>\n",
       "      <td>NM_001199973.1</td>\n",
       "      <td>1.14</td>\n",
       "      <td>511</td>\n",
       "      <td>46</td>\n",
       "      <td>465</td>\n",
       "      <td>532</td>\n",
       "      <td>413</td>\n",
       "      <td>0.89</td>\n",
       "      <td>246</td>\n",
       "      <td>0.53</td>\n",
       "      <td>602</td>\n",
       "      <td>1.29</td>\n",
       "      <td>965</td>\n",
       "      <td>2.07</td>\n",
       "      <td>176</td>\n",
       "      <td>0.38</td>\n",
       "      <td>206</td>\n",
       "      <td>0.44</td>\n",
       "      <td>185</td>\n",
       "      <td>0.40</td>\n",
       "      <td>809</td>\n",
       "      <td>1.74</td>\n",
       "      <td>551</td>\n",
       "      <td>1.18</td>\n",
       "      <td>436</td>\n",
       "      <td>0.94</td>\n",
       "      <td>659</td>\n",
       "      <td>1.42</td>\n",
       "      <td>686</td>\n",
       "      <td>1.48</td>\n",
       "      <td>540</td>\n",
       "      <td>1.16</td>\n",
       "      <td>873</td>\n",
       "      <td>1.88</td>\n",
       "      <td>RPL36A-HNRNPH2 readthrough (RPL36A-HNRNPH2), t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS20</th>\n",
       "      <td>NM_001146227.1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>627</td>\n",
       "      <td>198</td>\n",
       "      <td>429</td>\n",
       "      <td>427</td>\n",
       "      <td>409</td>\n",
       "      <td>0.95</td>\n",
       "      <td>311</td>\n",
       "      <td>0.72</td>\n",
       "      <td>525</td>\n",
       "      <td>1.22</td>\n",
       "      <td>800</td>\n",
       "      <td>1.86</td>\n",
       "      <td>119</td>\n",
       "      <td>0.28</td>\n",
       "      <td>152</td>\n",
       "      <td>0.35</td>\n",
       "      <td>123</td>\n",
       "      <td>0.29</td>\n",
       "      <td>699</td>\n",
       "      <td>1.63</td>\n",
       "      <td>562</td>\n",
       "      <td>1.31</td>\n",
       "      <td>672</td>\n",
       "      <td>1.57</td>\n",
       "      <td>634</td>\n",
       "      <td>1.48</td>\n",
       "      <td>623</td>\n",
       "      <td>1.45</td>\n",
       "      <td>511</td>\n",
       "      <td>1.19</td>\n",
       "      <td>801</td>\n",
       "      <td>1.87</td>\n",
       "      <td>ribosomal protein S20 (RPS20), transcript vari...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS12</th>\n",
       "      <td>NM_001016.3</td>\n",
       "      <td>0.89</td>\n",
       "      <td>481</td>\n",
       "      <td>82</td>\n",
       "      <td>399</td>\n",
       "      <td>357</td>\n",
       "      <td>257</td>\n",
       "      <td>0.64</td>\n",
       "      <td>214</td>\n",
       "      <td>0.54</td>\n",
       "      <td>480</td>\n",
       "      <td>1.20</td>\n",
       "      <td>714</td>\n",
       "      <td>1.79</td>\n",
       "      <td>88</td>\n",
       "      <td>0.22</td>\n",
       "      <td>126</td>\n",
       "      <td>0.32</td>\n",
       "      <td>83</td>\n",
       "      <td>0.21</td>\n",
       "      <td>631</td>\n",
       "      <td>1.58</td>\n",
       "      <td>535</td>\n",
       "      <td>1.34</td>\n",
       "      <td>444</td>\n",
       "      <td>1.11</td>\n",
       "      <td>521</td>\n",
       "      <td>1.31</td>\n",
       "      <td>428</td>\n",
       "      <td>1.07</td>\n",
       "      <td>351</td>\n",
       "      <td>0.88</td>\n",
       "      <td>672</td>\n",
       "      <td>1.68</td>\n",
       "      <td>ribosomal protein S12 (RPS12), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTRNR2L8</th>\n",
       "      <td>NM_001190702.1</td>\n",
       "      <td>2.89</td>\n",
       "      <td>1025</td>\n",
       "      <td>950</td>\n",
       "      <td>75</td>\n",
       "      <td>217</td>\n",
       "      <td>196</td>\n",
       "      <td>2.61</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>3</td>\n",
       "      <td>0.04</td>\n",
       "      <td>137</td>\n",
       "      <td>1.83</td>\n",
       "      <td>403</td>\n",
       "      <td>5.38</td>\n",
       "      <td>361</td>\n",
       "      <td>4.82</td>\n",
       "      <td>249</td>\n",
       "      <td>3.32</td>\n",
       "      <td>624</td>\n",
       "      <td>8.31</td>\n",
       "      <td>417</td>\n",
       "      <td>5.56</td>\n",
       "      <td>2</td>\n",
       "      <td>0.03</td>\n",
       "      <td>350</td>\n",
       "      <td>4.67</td>\n",
       "      <td>527</td>\n",
       "      <td>7.03</td>\n",
       "      <td>445</td>\n",
       "      <td>5.94</td>\n",
       "      <td>214</td>\n",
       "      <td>2.85</td>\n",
       "      <td>MT-RNR2-like 8 (MTRNR2L8), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS27</th>\n",
       "      <td>NM_001030.4</td>\n",
       "      <td>1.49</td>\n",
       "      <td>289</td>\n",
       "      <td>34</td>\n",
       "      <td>255</td>\n",
       "      <td>379</td>\n",
       "      <td>277</td>\n",
       "      <td>1.09</td>\n",
       "      <td>263</td>\n",
       "      <td>1.03</td>\n",
       "      <td>892</td>\n",
       "      <td>3.50</td>\n",
       "      <td>816</td>\n",
       "      <td>3.20</td>\n",
       "      <td>78</td>\n",
       "      <td>0.31</td>\n",
       "      <td>127</td>\n",
       "      <td>0.50</td>\n",
       "      <td>141</td>\n",
       "      <td>0.55</td>\n",
       "      <td>601</td>\n",
       "      <td>2.36</td>\n",
       "      <td>545</td>\n",
       "      <td>2.14</td>\n",
       "      <td>815</td>\n",
       "      <td>3.20</td>\n",
       "      <td>637</td>\n",
       "      <td>2.50</td>\n",
       "      <td>451</td>\n",
       "      <td>1.77</td>\n",
       "      <td>420</td>\n",
       "      <td>1.65</td>\n",
       "      <td>779</td>\n",
       "      <td>3.05</td>\n",
       "      <td>ribosomal protein S27 (RPS27), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TIMM8B</th>\n",
       "      <td>NM_012459.2</td>\n",
       "      <td>0.51</td>\n",
       "      <td>327</td>\n",
       "      <td>30</td>\n",
       "      <td>297</td>\n",
       "      <td>152</td>\n",
       "      <td>170</td>\n",
       "      <td>0.57</td>\n",
       "      <td>91</td>\n",
       "      <td>0.31</td>\n",
       "      <td>118</td>\n",
       "      <td>0.40</td>\n",
       "      <td>115</td>\n",
       "      <td>0.39</td>\n",
       "      <td>159</td>\n",
       "      <td>0.54</td>\n",
       "      <td>162</td>\n",
       "      <td>0.55</td>\n",
       "      <td>99</td>\n",
       "      <td>0.33</td>\n",
       "      <td>249</td>\n",
       "      <td>0.84</td>\n",
       "      <td>174</td>\n",
       "      <td>0.59</td>\n",
       "      <td>96</td>\n",
       "      <td>0.32</td>\n",
       "      <td>104</td>\n",
       "      <td>0.35</td>\n",
       "      <td>222</td>\n",
       "      <td>0.75</td>\n",
       "      <td>146</td>\n",
       "      <td>0.49</td>\n",
       "      <td>108</td>\n",
       "      <td>0.36</td>\n",
       "      <td>translocase of inner mitochondrial membrane 8 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIGD2A</th>\n",
       "      <td>NM_138820.2</td>\n",
       "      <td>0.43</td>\n",
       "      <td>358</td>\n",
       "      <td>37</td>\n",
       "      <td>321</td>\n",
       "      <td>139</td>\n",
       "      <td>91</td>\n",
       "      <td>0.28</td>\n",
       "      <td>72</td>\n",
       "      <td>0.22</td>\n",
       "      <td>123</td>\n",
       "      <td>0.38</td>\n",
       "      <td>123</td>\n",
       "      <td>0.38</td>\n",
       "      <td>76</td>\n",
       "      <td>0.24</td>\n",
       "      <td>95</td>\n",
       "      <td>0.30</td>\n",
       "      <td>95</td>\n",
       "      <td>0.30</td>\n",
       "      <td>238</td>\n",
       "      <td>0.74</td>\n",
       "      <td>98</td>\n",
       "      <td>0.31</td>\n",
       "      <td>86</td>\n",
       "      <td>0.27</td>\n",
       "      <td>115</td>\n",
       "      <td>0.36</td>\n",
       "      <td>189</td>\n",
       "      <td>0.59</td>\n",
       "      <td>108</td>\n",
       "      <td>0.34</td>\n",
       "      <td>105</td>\n",
       "      <td>0.33</td>\n",
       "      <td>HIG1 hypoxia inducible domain family, member 2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H2AE</th>\n",
       "      <td>NM_021052.2</td>\n",
       "      <td>0.30</td>\n",
       "      <td>448</td>\n",
       "      <td>55</td>\n",
       "      <td>393</td>\n",
       "      <td>116</td>\n",
       "      <td>87</td>\n",
       "      <td>0.22</td>\n",
       "      <td>111</td>\n",
       "      <td>0.28</td>\n",
       "      <td>67</td>\n",
       "      <td>0.17</td>\n",
       "      <td>51</td>\n",
       "      <td>0.13</td>\n",
       "      <td>67</td>\n",
       "      <td>0.17</td>\n",
       "      <td>87</td>\n",
       "      <td>0.22</td>\n",
       "      <td>63</td>\n",
       "      <td>0.16</td>\n",
       "      <td>191</td>\n",
       "      <td>0.49</td>\n",
       "      <td>166</td>\n",
       "      <td>0.42</td>\n",
       "      <td>80</td>\n",
       "      <td>0.20</td>\n",
       "      <td>69</td>\n",
       "      <td>0.18</td>\n",
       "      <td>158</td>\n",
       "      <td>0.40</td>\n",
       "      <td>128</td>\n",
       "      <td>0.33</td>\n",
       "      <td>57</td>\n",
       "      <td>0.15</td>\n",
       "      <td>histone cluster 1, H2ae (HIST1H2AE), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RRAGA</th>\n",
       "      <td>NM_006570.4</td>\n",
       "      <td>0.11</td>\n",
       "      <td>1228</td>\n",
       "      <td>286</td>\n",
       "      <td>942</td>\n",
       "      <td>102</td>\n",
       "      <td>120</td>\n",
       "      <td>0.13</td>\n",
       "      <td>108</td>\n",
       "      <td>0.11</td>\n",
       "      <td>86</td>\n",
       "      <td>0.09</td>\n",
       "      <td>107</td>\n",
       "      <td>0.11</td>\n",
       "      <td>129</td>\n",
       "      <td>0.14</td>\n",
       "      <td>141</td>\n",
       "      <td>0.15</td>\n",
       "      <td>162</td>\n",
       "      <td>0.17</td>\n",
       "      <td>180</td>\n",
       "      <td>0.19</td>\n",
       "      <td>189</td>\n",
       "      <td>0.20</td>\n",
       "      <td>316</td>\n",
       "      <td>0.34</td>\n",
       "      <td>177</td>\n",
       "      <td>0.19</td>\n",
       "      <td>229</td>\n",
       "      <td>0.24</td>\n",
       "      <td>225</td>\n",
       "      <td>0.24</td>\n",
       "      <td>192</td>\n",
       "      <td>0.20</td>\n",
       "      <td>Ras-related GTP binding A (RRAGA), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRWD1</th>\n",
       "      <td>NM_031485.3</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1574</td>\n",
       "      <td>233</td>\n",
       "      <td>1341</td>\n",
       "      <td>112</td>\n",
       "      <td>136</td>\n",
       "      <td>0.10</td>\n",
       "      <td>130</td>\n",
       "      <td>0.10</td>\n",
       "      <td>175</td>\n",
       "      <td>0.13</td>\n",
       "      <td>137</td>\n",
       "      <td>0.10</td>\n",
       "      <td>107</td>\n",
       "      <td>0.08</td>\n",
       "      <td>118</td>\n",
       "      <td>0.09</td>\n",
       "      <td>124</td>\n",
       "      <td>0.09</td>\n",
       "      <td>177</td>\n",
       "      <td>0.13</td>\n",
       "      <td>227</td>\n",
       "      <td>0.17</td>\n",
       "      <td>208</td>\n",
       "      <td>0.16</td>\n",
       "      <td>131</td>\n",
       "      <td>0.10</td>\n",
       "      <td>171</td>\n",
       "      <td>0.13</td>\n",
       "      <td>119</td>\n",
       "      <td>0.09</td>\n",
       "      <td>124</td>\n",
       "      <td>0.09</td>\n",
       "      <td>glutamate-rich WD repeat containing 1 (GRWD1),...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTRNR2L1</th>\n",
       "      <td>NM_001190452.1</td>\n",
       "      <td>1.23</td>\n",
       "      <td>1026</td>\n",
       "      <td>951</td>\n",
       "      <td>75</td>\n",
       "      <td>92</td>\n",
       "      <td>66</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>37</td>\n",
       "      <td>0.49</td>\n",
       "      <td>82</td>\n",
       "      <td>1.09</td>\n",
       "      <td>82</td>\n",
       "      <td>1.09</td>\n",
       "      <td>63</td>\n",
       "      <td>0.84</td>\n",
       "      <td>167</td>\n",
       "      <td>2.23</td>\n",
       "      <td>160</td>\n",
       "      <td>2.13</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>101</td>\n",
       "      <td>1.35</td>\n",
       "      <td>160</td>\n",
       "      <td>2.13</td>\n",
       "      <td>101</td>\n",
       "      <td>1.35</td>\n",
       "      <td>73</td>\n",
       "      <td>0.97</td>\n",
       "      <td>MT-RNR2-like 1 (MTRNR2L1), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTRNR2L2</th>\n",
       "      <td>NM_001190470.1</td>\n",
       "      <td>0.59</td>\n",
       "      <td>1036</td>\n",
       "      <td>949</td>\n",
       "      <td>87</td>\n",
       "      <td>51</td>\n",
       "      <td>75</td>\n",
       "      <td>0.86</td>\n",
       "      <td>2</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>53</td>\n",
       "      <td>0.61</td>\n",
       "      <td>98</td>\n",
       "      <td>1.13</td>\n",
       "      <td>84</td>\n",
       "      <td>0.97</td>\n",
       "      <td>61</td>\n",
       "      <td>0.70</td>\n",
       "      <td>156</td>\n",
       "      <td>1.79</td>\n",
       "      <td>134</td>\n",
       "      <td>1.54</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>99</td>\n",
       "      <td>1.14</td>\n",
       "      <td>150</td>\n",
       "      <td>1.72</td>\n",
       "      <td>110</td>\n",
       "      <td>1.26</td>\n",
       "      <td>88</td>\n",
       "      <td>1.01</td>\n",
       "      <td>MT-RNR2-like 2 (MTRNR2L2), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BOLA2B</th>\n",
       "      <td>NM_001039182.1</td>\n",
       "      <td>0.21</td>\n",
       "      <td>972</td>\n",
       "      <td>513</td>\n",
       "      <td>459</td>\n",
       "      <td>98</td>\n",
       "      <td>152</td>\n",
       "      <td>0.33</td>\n",
       "      <td>56</td>\n",
       "      <td>0.12</td>\n",
       "      <td>78</td>\n",
       "      <td>0.17</td>\n",
       "      <td>107</td>\n",
       "      <td>0.23</td>\n",
       "      <td>71</td>\n",
       "      <td>0.15</td>\n",
       "      <td>105</td>\n",
       "      <td>0.23</td>\n",
       "      <td>60</td>\n",
       "      <td>0.13</td>\n",
       "      <td>145</td>\n",
       "      <td>0.32</td>\n",
       "      <td>142</td>\n",
       "      <td>0.31</td>\n",
       "      <td>48</td>\n",
       "      <td>0.10</td>\n",
       "      <td>115</td>\n",
       "      <td>0.25</td>\n",
       "      <td>143</td>\n",
       "      <td>0.31</td>\n",
       "      <td>146</td>\n",
       "      <td>0.32</td>\n",
       "      <td>139</td>\n",
       "      <td>0.30</td>\n",
       "      <td>bolA homolog 2B (E. coli) (BOLA2B), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID1</th>\n",
       "      <td>NM_002165.3</td>\n",
       "      <td>0.19</td>\n",
       "      <td>573</td>\n",
       "      <td>105</td>\n",
       "      <td>468</td>\n",
       "      <td>89</td>\n",
       "      <td>78</td>\n",
       "      <td>0.17</td>\n",
       "      <td>62</td>\n",
       "      <td>0.13</td>\n",
       "      <td>52</td>\n",
       "      <td>0.11</td>\n",
       "      <td>35</td>\n",
       "      <td>0.07</td>\n",
       "      <td>96</td>\n",
       "      <td>0.21</td>\n",
       "      <td>97</td>\n",
       "      <td>0.21</td>\n",
       "      <td>102</td>\n",
       "      <td>0.22</td>\n",
       "      <td>140</td>\n",
       "      <td>0.30</td>\n",
       "      <td>91</td>\n",
       "      <td>0.19</td>\n",
       "      <td>61</td>\n",
       "      <td>0.13</td>\n",
       "      <td>54</td>\n",
       "      <td>0.11</td>\n",
       "      <td>111</td>\n",
       "      <td>0.24</td>\n",
       "      <td>115</td>\n",
       "      <td>0.25</td>\n",
       "      <td>62</td>\n",
       "      <td>0.13</td>\n",
       "      <td>inhibitor of DNA binding 1, dominant negative ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EMG1</th>\n",
       "      <td>NM_006331.7</td>\n",
       "      <td>0.10</td>\n",
       "      <td>878</td>\n",
       "      <td>143</td>\n",
       "      <td>735</td>\n",
       "      <td>77</td>\n",
       "      <td>96</td>\n",
       "      <td>0.13</td>\n",
       "      <td>93</td>\n",
       "      <td>0.13</td>\n",
       "      <td>126</td>\n",
       "      <td>0.17</td>\n",
       "      <td>119</td>\n",
       "      <td>0.16</td>\n",
       "      <td>133</td>\n",
       "      <td>0.18</td>\n",
       "      <td>122</td>\n",
       "      <td>0.17</td>\n",
       "      <td>111</td>\n",
       "      <td>0.15</td>\n",
       "      <td>133</td>\n",
       "      <td>0.18</td>\n",
       "      <td>106</td>\n",
       "      <td>0.14</td>\n",
       "      <td>86</td>\n",
       "      <td>0.12</td>\n",
       "      <td>86</td>\n",
       "      <td>0.12</td>\n",
       "      <td>84</td>\n",
       "      <td>0.11</td>\n",
       "      <td>103</td>\n",
       "      <td>0.14</td>\n",
       "      <td>103</td>\n",
       "      <td>0.14</td>\n",
       "      <td>EMG1 N1-specific pseudouridine methyltransfera...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DUSP1</th>\n",
       "      <td>NM_004417.3</td>\n",
       "      <td>0.05</td>\n",
       "      <td>1352</td>\n",
       "      <td>248</td>\n",
       "      <td>1104</td>\n",
       "      <td>58</td>\n",
       "      <td>34</td>\n",
       "      <td>0.03</td>\n",
       "      <td>37</td>\n",
       "      <td>0.03</td>\n",
       "      <td>38</td>\n",
       "      <td>0.03</td>\n",
       "      <td>23</td>\n",
       "      <td>0.02</td>\n",
       "      <td>22</td>\n",
       "      <td>0.02</td>\n",
       "      <td>21</td>\n",
       "      <td>0.02</td>\n",
       "      <td>32</td>\n",
       "      <td>0.03</td>\n",
       "      <td>113</td>\n",
       "      <td>0.10</td>\n",
       "      <td>43</td>\n",
       "      <td>0.04</td>\n",
       "      <td>16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>18</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98</td>\n",
       "      <td>0.09</td>\n",
       "      <td>40</td>\n",
       "      <td>0.04</td>\n",
       "      <td>34</td>\n",
       "      <td>0.03</td>\n",
       "      <td>dual specificity phosphatase 1 (DUSP1), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL36</th>\n",
       "      <td>NM_032479.3</td>\n",
       "      <td>0.20</td>\n",
       "      <td>375</td>\n",
       "      <td>63</td>\n",
       "      <td>312</td>\n",
       "      <td>62</td>\n",
       "      <td>70</td>\n",
       "      <td>0.22</td>\n",
       "      <td>33</td>\n",
       "      <td>0.11</td>\n",
       "      <td>57</td>\n",
       "      <td>0.18</td>\n",
       "      <td>72</td>\n",
       "      <td>0.23</td>\n",
       "      <td>52</td>\n",
       "      <td>0.17</td>\n",
       "      <td>50</td>\n",
       "      <td>0.16</td>\n",
       "      <td>49</td>\n",
       "      <td>0.16</td>\n",
       "      <td>112</td>\n",
       "      <td>0.36</td>\n",
       "      <td>69</td>\n",
       "      <td>0.22</td>\n",
       "      <td>78</td>\n",
       "      <td>0.25</td>\n",
       "      <td>75</td>\n",
       "      <td>0.24</td>\n",
       "      <td>86</td>\n",
       "      <td>0.28</td>\n",
       "      <td>79</td>\n",
       "      <td>0.25</td>\n",
       "      <td>84</td>\n",
       "      <td>0.27</td>\n",
       "      <td>mitochondrial ribosomal protein L36 (MRPL36), ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCAND1</th>\n",
       "      <td>NM_016558.3</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1243</td>\n",
       "      <td>703</td>\n",
       "      <td>540</td>\n",
       "      <td>67</td>\n",
       "      <td>42</td>\n",
       "      <td>0.08</td>\n",
       "      <td>59</td>\n",
       "      <td>0.11</td>\n",
       "      <td>84</td>\n",
       "      <td>0.16</td>\n",
       "      <td>39</td>\n",
       "      <td>0.07</td>\n",
       "      <td>28</td>\n",
       "      <td>0.05</td>\n",
       "      <td>53</td>\n",
       "      <td>0.10</td>\n",
       "      <td>45</td>\n",
       "      <td>0.08</td>\n",
       "      <td>106</td>\n",
       "      <td>0.20</td>\n",
       "      <td>50</td>\n",
       "      <td>0.09</td>\n",
       "      <td>69</td>\n",
       "      <td>0.13</td>\n",
       "      <td>31</td>\n",
       "      <td>0.06</td>\n",
       "      <td>125</td>\n",
       "      <td>0.23</td>\n",
       "      <td>54</td>\n",
       "      <td>0.10</td>\n",
       "      <td>65</td>\n",
       "      <td>0.12</td>\n",
       "      <td>SCAN domain containing 1 (SCAND1), transcript ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ELOF1</th>\n",
       "      <td>NM_032377.3</td>\n",
       "      <td>0.25</td>\n",
       "      <td>316</td>\n",
       "      <td>64</td>\n",
       "      <td>252</td>\n",
       "      <td>64</td>\n",
       "      <td>65</td>\n",
       "      <td>0.26</td>\n",
       "      <td>31</td>\n",
       "      <td>0.12</td>\n",
       "      <td>53</td>\n",
       "      <td>0.21</td>\n",
       "      <td>63</td>\n",
       "      <td>0.25</td>\n",
       "      <td>43</td>\n",
       "      <td>0.17</td>\n",
       "      <td>48</td>\n",
       "      <td>0.19</td>\n",
       "      <td>38</td>\n",
       "      <td>0.15</td>\n",
       "      <td>106</td>\n",
       "      <td>0.42</td>\n",
       "      <td>70</td>\n",
       "      <td>0.28</td>\n",
       "      <td>45</td>\n",
       "      <td>0.18</td>\n",
       "      <td>68</td>\n",
       "      <td>0.27</td>\n",
       "      <td>97</td>\n",
       "      <td>0.38</td>\n",
       "      <td>77</td>\n",
       "      <td>0.31</td>\n",
       "      <td>72</td>\n",
       "      <td>0.29</td>\n",
       "      <td>elongation factor 1 homolog (S. cerevisiae) (E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRRC2B</th>\n",
       "      <td>NM_013318.3</td>\n",
       "      <td>0.01</td>\n",
       "      <td>6745</td>\n",
       "      <td>55</td>\n",
       "      <td>6690</td>\n",
       "      <td>94</td>\n",
       "      <td>81</td>\n",
       "      <td>0.01</td>\n",
       "      <td>148</td>\n",
       "      <td>0.02</td>\n",
       "      <td>102</td>\n",
       "      <td>0.02</td>\n",
       "      <td>90</td>\n",
       "      <td>0.01</td>\n",
       "      <td>102</td>\n",
       "      <td>0.02</td>\n",
       "      <td>121</td>\n",
       "      <td>0.02</td>\n",
       "      <td>114</td>\n",
       "      <td>0.02</td>\n",
       "      <td>102</td>\n",
       "      <td>0.02</td>\n",
       "      <td>113</td>\n",
       "      <td>0.02</td>\n",
       "      <td>134</td>\n",
       "      <td>0.02</td>\n",
       "      <td>83</td>\n",
       "      <td>0.01</td>\n",
       "      <td>87</td>\n",
       "      <td>0.01</td>\n",
       "      <td>103</td>\n",
       "      <td>0.02</td>\n",
       "      <td>99</td>\n",
       "      <td>0.01</td>\n",
       "      <td>proline-rich coiled-coil 2B (PRRC2B), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H2BG</th>\n",
       "      <td>NM_003518.3</td>\n",
       "      <td>0.14</td>\n",
       "      <td>382</td>\n",
       "      <td>1</td>\n",
       "      <td>381</td>\n",
       "      <td>54</td>\n",
       "      <td>28</td>\n",
       "      <td>0.07</td>\n",
       "      <td>36</td>\n",
       "      <td>0.09</td>\n",
       "      <td>72</td>\n",
       "      <td>0.19</td>\n",
       "      <td>23</td>\n",
       "      <td>0.06</td>\n",
       "      <td>35</td>\n",
       "      <td>0.09</td>\n",
       "      <td>29</td>\n",
       "      <td>0.08</td>\n",
       "      <td>38</td>\n",
       "      <td>0.10</td>\n",
       "      <td>88</td>\n",
       "      <td>0.23</td>\n",
       "      <td>73</td>\n",
       "      <td>0.19</td>\n",
       "      <td>38</td>\n",
       "      <td>0.10</td>\n",
       "      <td>43</td>\n",
       "      <td>0.11</td>\n",
       "      <td>73</td>\n",
       "      <td>0.19</td>\n",
       "      <td>53</td>\n",
       "      <td>0.14</td>\n",
       "      <td>35</td>\n",
       "      <td>0.09</td>\n",
       "      <td>histone cluster 1, H2bg (HIST1H2BG), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WDR73</th>\n",
       "      <td>NM_032856.2</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1153</td>\n",
       "      <td>16</td>\n",
       "      <td>1137</td>\n",
       "      <td>51</td>\n",
       "      <td>61</td>\n",
       "      <td>0.05</td>\n",
       "      <td>50</td>\n",
       "      <td>0.04</td>\n",
       "      <td>81</td>\n",
       "      <td>0.07</td>\n",
       "      <td>49</td>\n",
       "      <td>0.04</td>\n",
       "      <td>65</td>\n",
       "      <td>0.06</td>\n",
       "      <td>56</td>\n",
       "      <td>0.05</td>\n",
       "      <td>53</td>\n",
       "      <td>0.05</td>\n",
       "      <td>88</td>\n",
       "      <td>0.08</td>\n",
       "      <td>50</td>\n",
       "      <td>0.04</td>\n",
       "      <td>55</td>\n",
       "      <td>0.05</td>\n",
       "      <td>56</td>\n",
       "      <td>0.05</td>\n",
       "      <td>58</td>\n",
       "      <td>0.05</td>\n",
       "      <td>45</td>\n",
       "      <td>0.04</td>\n",
       "      <td>44</td>\n",
       "      <td>0.04</td>\n",
       "      <td>WD repeat domain 73 (WDR73), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CBX6</th>\n",
       "      <td>NM_014292.3</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1302</td>\n",
       "      <td>63</td>\n",
       "      <td>1239</td>\n",
       "      <td>53</td>\n",
       "      <td>67</td>\n",
       "      <td>0.05</td>\n",
       "      <td>88</td>\n",
       "      <td>0.07</td>\n",
       "      <td>145</td>\n",
       "      <td>0.12</td>\n",
       "      <td>89</td>\n",
       "      <td>0.07</td>\n",
       "      <td>51</td>\n",
       "      <td>0.04</td>\n",
       "      <td>51</td>\n",
       "      <td>0.04</td>\n",
       "      <td>61</td>\n",
       "      <td>0.05</td>\n",
       "      <td>85</td>\n",
       "      <td>0.07</td>\n",
       "      <td>114</td>\n",
       "      <td>0.09</td>\n",
       "      <td>53</td>\n",
       "      <td>0.04</td>\n",
       "      <td>134</td>\n",
       "      <td>0.11</td>\n",
       "      <td>67</td>\n",
       "      <td>0.05</td>\n",
       "      <td>64</td>\n",
       "      <td>0.05</td>\n",
       "      <td>120</td>\n",
       "      <td>0.10</td>\n",
       "      <td>chromobox homolog 6 (CBX6), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPS12</th>\n",
       "      <td>NM_021107.1</td>\n",
       "      <td>0.13</td>\n",
       "      <td>758</td>\n",
       "      <td>341</td>\n",
       "      <td>417</td>\n",
       "      <td>56</td>\n",
       "      <td>42</td>\n",
       "      <td>0.10</td>\n",
       "      <td>41</td>\n",
       "      <td>0.10</td>\n",
       "      <td>91</td>\n",
       "      <td>0.22</td>\n",
       "      <td>65</td>\n",
       "      <td>0.16</td>\n",
       "      <td>46</td>\n",
       "      <td>0.11</td>\n",
       "      <td>56</td>\n",
       "      <td>0.13</td>\n",
       "      <td>42</td>\n",
       "      <td>0.10</td>\n",
       "      <td>83</td>\n",
       "      <td>0.20</td>\n",
       "      <td>66</td>\n",
       "      <td>0.16</td>\n",
       "      <td>59</td>\n",
       "      <td>0.14</td>\n",
       "      <td>42</td>\n",
       "      <td>0.10</td>\n",
       "      <td>69</td>\n",
       "      <td>0.17</td>\n",
       "      <td>54</td>\n",
       "      <td>0.13</td>\n",
       "      <td>54</td>\n",
       "      <td>0.13</td>\n",
       "      <td>mitochondrial ribosomal protein S12 (MRPS12), ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SZRD1</th>\n",
       "      <td>NM_001114600.2</td>\n",
       "      <td>0.11</td>\n",
       "      <td>687</td>\n",
       "      <td>228</td>\n",
       "      <td>459</td>\n",
       "      <td>52</td>\n",
       "      <td>59</td>\n",
       "      <td>0.13</td>\n",
       "      <td>20</td>\n",
       "      <td>0.04</td>\n",
       "      <td>42</td>\n",
       "      <td>0.09</td>\n",
       "      <td>46</td>\n",
       "      <td>0.10</td>\n",
       "      <td>77</td>\n",
       "      <td>0.17</td>\n",
       "      <td>78</td>\n",
       "      <td>0.17</td>\n",
       "      <td>41</td>\n",
       "      <td>0.09</td>\n",
       "      <td>80</td>\n",
       "      <td>0.17</td>\n",
       "      <td>69</td>\n",
       "      <td>0.15</td>\n",
       "      <td>33</td>\n",
       "      <td>0.07</td>\n",
       "      <td>51</td>\n",
       "      <td>0.11</td>\n",
       "      <td>79</td>\n",
       "      <td>0.17</td>\n",
       "      <td>60</td>\n",
       "      <td>0.13</td>\n",
       "      <td>53</td>\n",
       "      <td>0.12</td>\n",
       "      <td>SUZ RNA binding domain containing 1 (SZRD1), t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LARP4B</th>\n",
       "      <td>NM_015155.2</td>\n",
       "      <td>0.02</td>\n",
       "      <td>2534</td>\n",
       "      <td>317</td>\n",
       "      <td>2217</td>\n",
       "      <td>53</td>\n",
       "      <td>71</td>\n",
       "      <td>0.03</td>\n",
       "      <td>68</td>\n",
       "      <td>0.03</td>\n",
       "      <td>44</td>\n",
       "      <td>0.02</td>\n",
       "      <td>75</td>\n",
       "      <td>0.03</td>\n",
       "      <td>75</td>\n",
       "      <td>0.03</td>\n",
       "      <td>73</td>\n",
       "      <td>0.03</td>\n",
       "      <td>101</td>\n",
       "      <td>0.05</td>\n",
       "      <td>78</td>\n",
       "      <td>0.04</td>\n",
       "      <td>83</td>\n",
       "      <td>0.04</td>\n",
       "      <td>77</td>\n",
       "      <td>0.03</td>\n",
       "      <td>73</td>\n",
       "      <td>0.03</td>\n",
       "      <td>72</td>\n",
       "      <td>0.03</td>\n",
       "      <td>73</td>\n",
       "      <td>0.03</td>\n",
       "      <td>80</td>\n",
       "      <td>0.04</td>\n",
       "      <td>La ribonucleoprotein domain family, member 4B ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POGZ</th>\n",
       "      <td>NM_015100.3</td>\n",
       "      <td>0.01</td>\n",
       "      <td>4548</td>\n",
       "      <td>315</td>\n",
       "      <td>4233</td>\n",
       "      <td>62</td>\n",
       "      <td>81</td>\n",
       "      <td>0.02</td>\n",
       "      <td>85</td>\n",
       "      <td>0.02</td>\n",
       "      <td>42</td>\n",
       "      <td>0.01</td>\n",
       "      <td>47</td>\n",
       "      <td>0.01</td>\n",
       "      <td>102</td>\n",
       "      <td>0.02</td>\n",
       "      <td>89</td>\n",
       "      <td>0.02</td>\n",
       "      <td>99</td>\n",
       "      <td>0.02</td>\n",
       "      <td>73</td>\n",
       "      <td>0.02</td>\n",
       "      <td>53</td>\n",
       "      <td>0.01</td>\n",
       "      <td>77</td>\n",
       "      <td>0.02</td>\n",
       "      <td>74</td>\n",
       "      <td>0.02</td>\n",
       "      <td>68</td>\n",
       "      <td>0.02</td>\n",
       "      <td>67</td>\n",
       "      <td>0.02</td>\n",
       "      <td>52</td>\n",
       "      <td>0.01</td>\n",
       "      <td>pogo transposable element with ZNF domain (POG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.core.display import HTML\n",
    "#temp = data[subset]\n",
    "HTML( data[ numpy.logical_and((data['rpnt_ragbq99l_arg'] / data['rpnt_hrgfp_arg']) > 1.5, data['counts_hrgfp_arg'] > 50 )].sort(columns=['counts_ragbq99l_arg'],ascending=False).to_html() )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
